### PR TITLE
Implement LpNormalization operator (#134)

### LINKED ISSUE #134: Implement ONNX LpNormalization operator support
### Feature description

Implement full support for the ONNX `LpNormalization` operator in the Burn ONNX import pipeline.

This requires:
- Adding parsing for the `LpNormalization` node and all its attributes in `onnx-ir`, ensuring *all* ONNX attributes are faithfully extracted, including optional ones and opset/version deltas.
- Implementing a config struct in `onnx-ir` that mirrors ONNX semantics (do not translate or pre-compute any attributes — import ONNX as-is).
- In `burn-onnx`, add code generation for `LpNormalization` via the `NodeCodegen` trait. This should translate ONNX semantics to Burn API calls in the most idiomatic and efficient way.
- If the Burn API doesn't (yet) support `LpNormalization`, emit a clear, actionable error at codegen time.
- Add comprehensive `insta` snapshot tests for all codegen branches (different config variants, opset differences, etc.).

### Feature motivation

`LpNormalization` is a common operator used for normalizing input features with respect to an Lp-norm. Its absence prevents importing a range of real-world ONNX models.

Adding complete support will:
- Improve opset coverage and compatibility
- Enable importing models that require feature normalization
- Move us closer to full ONNX compliance

### (Optional) Suggest a Solution

- Start from the [ONNX operator spec](https://onnx.ai/onnx/operators/onnx__LpNormalization.html), especially attribute details and opset version history
- Look at existing onnx-ir node processors for standard extraction patterns
- Ensure the onnx-ir config includes *all* operator attributes, even if not immediately used
- For codegen, resolve ONNX vs Burn API semantic differences here only
- Provide snapshot/coverage tests for each config combination and edge case

### PR DESCRIPTION
**Pull request by antimora**
## Summary
- Adds ONNX `LpNormalization` support end-to-end: onnx-ir parsing with attribute validation (`p` restricted to 1 or 2 per spec, negative-axis resolution, opset-aware dtype gating including bf16 at opset 22+), burn-onnx codegen delegating to `burn::tensor::linalg::l1_norm`/`l2_norm`, integration tests, opset-compliance tests, and enables the 6 official ONNX Lp tests that were previously in `skip-codegen`.
- Closes #134.

## Notes
- Integration tests use ONNX Runtime as ground truth rather than `onnx.reference.ReferenceEvaluator`: the reference evaluator (as of `onnx==1.19.0`) computes `x^p` instead of `|x|^p`, producing wrong results for `p=1` on signed inputs. ORT and PyTorch's `F.normalize` both use the correct `|x|^p` form.
- Zero-norm slices produce NaN/Inf, matching ORT. The reference evaluator's `where(norm==0, 0, ...)` guard is not part of the ONNX spec.

## Test plan
- [x] `cargo test -p onnx-ir` — 10 unit tests for `LpNormalizationProcessor` (defaults, negative axes including the `-rank` boundary, invalid attrs, dtype gating, bf16 at opset <22 vs >=22)
- [x] `cargo test -p burn-onnx lp_normalization` — 4 snapshot tests (p=1, p=2, non-trailing axis, multi-consumer clone)
- [x] `cargo test -p onnx-tests lp_normalization` — 4 integration tests (default, L1/axis=1, L2/axis=0, L2/negative-axis), expected outputs from ORT
- [x] `cargo test -p onnx-ir --test opset_compliance` — 469 tests pass including new `opset_01::lp_normalization` and `opset_22::lp_normalization`
- [x] `cargo test -p onnx-official-tests` — 595 tests pass; promoted `test_l1normalization_axis_0/1/last`, `test_l2normalization_axis_0/1`, `test_lpnormalization_default` from `skip-codegen` to `pass`

### CODE REVIEW

File: crates/onnx-ir/src/node/lp_normalization.rs [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    fn infer_types(
+        &self,
+        node: &mut RawNode,
+        opset: usize,
+        _output_preferences: &OutputPreferences,
+    ) -> Result<(), ProcessError> {
+        let arg = node
+            .inputs
+            .first()
+            .ok_or_else(|| ProcessError::MissingInput("Missing input".to_string()))?;
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/lp_normalization.rs]: Good catch — fixed in 0fc0e521's follow-up. Used `"input"` to match the ONNX spec's name for this operator's input (which is `input`, not `X`), aligning with the pattern in `resize.rs`. This is more actionable than the generic `"Missing input"` I had, and stays faithful to the spec rather than adopting the conventional `X` name (which doesn't appear in the LpNormalization schema).
